### DISCUSSION TITLE
Does new Multi-object Cable Terminations not fully supported breakout cable?

### DISCUSSION THREAD

**Discussion by gamerslouis** [netbox-community/netbox#10182] [Q&A]
I notice that v3.3.0 add the the feature Multi-object Cable Terminations (https://github.com/netbox-community/netbox/issues/9102)
The description show that this can be used to modeling breakout cable.

So I try to create cable connection from one device interface to another "two devices" interfaces. 
But I found that the "connect cable ui still design to connect only between one device and another one.
It seems that I can't select two interfaces from two device at "B Side"
![image](https://user-images.githubusercontent.com/32395144/187085849-b5dd6b73-e11f-47e1-b615-f71b12d3cc2d.png)

However, if I select interfaces on one device and switch B Side device field to another device, then I can select interfaces from two device since the interfaces field is not cleared if the device field change.

After that I can get a weird trace graph and cable view
![image](https://user-images.githubusercontent.com/32395144/187086048-63b53fa9-7de5-4f85-bad4-fb7866ed7d37.png)
![image](https://user-images.githubusercontent.com/32395144/187086148-d15effb9-664e-4eb2-a832-66d301f79bd0.png)

In this case, termination B contains interfaces:
    - dmi01-akron-sw01 GigabitEthernet1/0/13
    - dmi01-akron-sw01 GigabitEthernet1/0/14
    - dmi01-akron-sw01 GigabitEthernet1/0/15
    - dmi01-camden-sw01 GigabitEthernet1/0/7
    - dmi01-camden-sw01 GigabitEthernet1/0/11

Then I try to remove akron GigabitEthernet1/0/13 and camden GigabitEthernet1/0/7 by edit cable then I get a AssertionError.

**Comment by jeremystretch** [comment]
> After that I can get a weird trace graph and cable view

This is a limitation of how the trace is being rendered. We'll to tweak the SVG rendering logic to clarify the device-to-component associations.

> Then I try to remove akron GigabitEthernet1/0/13 and camden GigabitEthernet1/0/7 by edit cable then I get a AssertionError.

This would be a separate issue. Please open a bug report if you're able to provide complete reproduction instructions.

**Reply by gamerslouis** [reply]
Thank you for your reply!

I think that there is another UI issue.

This cable connect to five interfaces belonging to two different device.
![image](https://user-images.githubusercontent.com/32395144/187722172-6078d8a8-8b2d-4fd3-9c09-03a76b7aa660.png)

But in cable page and connect cable page, it doesn't show these. It like that five interface all belong to single device.
![image](https://user-images.githubusercontent.com/32395144/187722739-ba43cb44-60ab-4a65-8eaa-e4c75940cc78.png)
![image](https://user-images.githubusercontent.com/32395144/187722821-5331e9ba-5056-4174-b8bb-2f7adb4bc392.png)

I think the template should be modified, and one possibility can be
![image](https://user-images.githubusercontent.com/32395144/187724459-0a3f48cd-9d62-479e-bad8-235048fb0045.png)


I am not sure if this match Netbox ui style and easy to identify.
Should I open a issue for this problem?

**Comment by Joelp** [comment]
This solution does not work in the latest version v4.0.6. Is multiobject cable terminations still possible?

**Reply by candlerb** [reply]
Which solution, and what is your problem? Yes, multiple cable terminations to different devices do work in v4.0

**Reply by Joelp** [reply]
can I ask you for a link to documentation, or hint how to do cable multi termination to different devices? Only I know is how to createcable with one device on A side and one device on B side.

I need to connect breakout cable to device A (40G) and 4 different devices on side B (10G).

**Reply by candlerb** [reply]
There's a description in #10841 (and discussions which reference that issue, e.g. #15175). However that issue says that the entry form has been changed.

A quick test with the entry form in 4.0.6 and I can't see how to make it work either; whenever you choose a new device, the list of interfaces is cleared.

I suggest you raise this as a bug. Give the exact set of steps you're trying to do, and screenshots before and after the interface list is cleared.

**Reply by jeremystretch** [reply]
Please see #16176. It's currently open for volunteers.
